Lib/_pyrepl/reader.py

```diff
@@ -473,22 +475,40 @@ def get_arg(self, default: int = 1) -> int:
             return default
         return self.arg
 
+    @staticmethod
+    def __get_prompt_str(prompt: object, default_prompt: str) -> str:
+        """
+        Convert prompt object to string.
+
+        If str(prompt) raises BaseException, MemoryError or SystemError then stop
```

Comment:
I think now this is accurate:
```suggestion
        If str(prompt) raises MemoryError or SystemError then stop
```

Reply:
Good catch, thanks!